crates/core/src/lib.rs

```diff
@@ -245,11 +245,47 @@ impl<T: Send + Sync> Engine<T> {
     #[instrument(skip_all)]
     pub fn instantiate_pre(&self, module: &Module) -> Result<InstancePre<T>> {
         let mut store = self.instance_pre_store.lock().unwrap();
-        let inner = self.linker.instantiate_pre(&mut *store, module)?;
+        let inner = self
+            .linker
+            .instantiate_pre(&mut *store, module)
+            .map_err(decode_preinstantiation_error)?;
         Ok(InstancePre { inner })
     }
 }
 
+fn decode_preinstantiation_error(e: anyhow::Error) -> anyhow::Error {
+    let err_text = e.to_string();
+
+    if err_text.contains("unknown import") && err_text.contains("has not been defined") {
+        // TODO: how to maintain this list?
```

Comment:
Yeah... then we'd need to have this somewhere it could consult the linker, and we'd still need the legacy list... I might punt on this for now.  The failure mode is that it still tells you which component has a problem but doesn't point fingers at the SDK, which is pretty benign.